src/generators/json/utils/createMethodSection.mjs

```diff
@@ -0,0 +1,356 @@
+) {
+  const parameterGroupings = createParameterGroupings(parameterNames);
+
+  let signatureIndex = section.signatures?.length ?? 0;
+  section.signatures = [
+    ...(section.signatures ?? []),
+    ...new Array(parameterGroupings.length),
+  ];
+
+  for (const grouping of parameterGroupings) {
+    /**
+     * @type {Array<import('../generated.d.ts').MethodParameter>}
+     */
+    const signatureParameters = new Array(grouping.length);
+
+    for (let i = 0; i < signatureParameters.length; i++) {
+      const parameterName = grouping[i];
+
+      // TODO handle default value for parameters
+
+      let parameter =
+        parameterName in parameters ? parameters[parameterName] : undefined;
+
+      if (!parameter) {
+        // console.warn(
+        //   `parameter name ${parameterName} included in method's signature but not parameter list, defaulting to type \`any\``
+        // );
+        parameter = { '@name': 'value', '@type': 'any' };
+      }
+
+      signatureParameters[i] = parameter;
+    }
+
+    section.signatures[signatureIndex] = {
+      '@returns': baseSignature['@returns'],
+      parameters: signatureParameters,
+    };
+
+    signatureIndex++;
+  }
+
+  // console.log(
+  //   'signatures',
+  //   util.inspect(section.signatures, false, null, true)
+  // );
+}
+
+/**
+ * Parses the signatures that the method may have and adds them to the
+ * section.
+ * @param {HierarchizedEntry} entry The AST entry
+ * @param {import('../generated.d.ts').Method} section The method section
+ */
+export function parseSignatures(entry, section) {
+  section.signatures = [];
+
+  /**
+   * @type {import('../generated.d.ts').MethodSignature}
+   */
+  const baseSignature = {
+    '@returns': { '@type': 'any' },
+  };
+
+  // Parse the parameters defined in the parameter list and grab the return
+  // type (if the list actually exists and those are present in it)
+  const parameterList = parseParameterList(entry) ?? {};
+
+  if (parameterList.returns) {
+    // Return type was defined in the parameter list, let's update the base
+    // signature
+    baseSignature['@returns'] = parameterList.returns;
+  }
+
+  /**
+   * Extract the parameter names mentioned in the entry's header. This gives
+   * us 1) the order in which they appear and 2) whether or not they're
+   * optional
+   * @example `[sources[, options]]`
+   */
+  let [, parametersNames] =
+    entry.heading.data.text
+      .substring(1, entry.heading.data.text.length - 1)
+      .match(METHOD_PARAM_EXPRESSION) || [];
+  if (!parametersNames || !parameterList.parameters) {
+    // Method doesn't have any parameters, return early
+    section.signatures.push(baseSignature);
+    return;
+  }
+
+  /**
+   * @example ['[sources[', 'options]]']`
+   */
+  parametersNames = parametersNames.split(',');
+
+  createSignatures(
+    section,
+    baseSignature,
+    parametersNames,
+    parameterList.parameters
+  );
+}
+
+/**
+ *
+ */
+export const createMethodSectionBuilder = () => {
```

Comment:
1. Can we put all these builders in `utils/sections/[class,event,method,module,propertty,base]`?
2. Do they really need to be `() => () => [...]`, it seems redundant to require a `create[...]SectionBuilder()` call to create the builder, can't the builder just be exported directly?